org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java

```diff
@@ -1349,6 +1356,9 @@ private Expression convertExpressionImpl(JCExpression javac) {
 				.map(JCVariableDecl.class::cast)
 				.map(this::convertVariableDeclarationForLambda)
 				.forEach(res.parameters()::add);
+			int arrowIndex = this.rawText.indexOf("->", jcLambda.getStartPosition());
+			int parenthesisIndex = this.rawText.indexOf(")", jcLambda.getStartPosition());
+			res.setParentheses(parenthesisIndex >= 0 && parenthesisIndex < arrowIndex);
```

Comment:
what if there's no opening ( ?

Reply:
Then we most likely receive a JCErroneous instead of a JCLambda